gnosis/safe-react
# Use SignMessageLib for signing

## What it solves
Resolves #2677 

## How this PR fixes it
- Adds a handler for `signMessage` RPC call from Safe Apps 
- Introduces a new component `SignMessageModal` which is displayed after the interface receives such a message
- If the user approves signing, a delegate call transaction is sent to `SignMessageLib`

## How to test it
Launch https://github.com/gnosis/safe-test-app, on the main screen at the bottom there should be an input for a message, type your message and sign it. Then click the check button to see if the message has been signed.
Some cases to consider:
1. Regular text
2. UTF-8 text converted to hex -> should see UTF-8 text in the modal
3. Transaction hash, address -> should see the hash, it cannot be converted to utf-8

## Screenshots
![image](https://user-images.githubusercontent.com/16622558/132683478-1fd2f585-1810-4569-957d-33494758d1e0.png)
